G1ANT.Addon.Images/G1ANT.Addon.Images/Api/Imaging.cs

```diff
@@ -46,97 +51,114 @@ public static bool AreImagesEqual(Bitmap image1, Bitmap image2)
 
         public static Rectangle IsImageInImage(Bitmap smallBmp, Bitmap bigBmp, double tolerance)
         {
-            BitmapData smallData =
-              smallBmp.LockBits(new Rectangle(0, 0, smallBmp.Width, smallBmp.Height),
-                       System.Drawing.Imaging.ImageLockMode.ReadOnly,
-                       System.Drawing.Imaging.PixelFormat.Format24bppRgb);
-            BitmapData bigData =
-              bigBmp.LockBits(new Rectangle(0, 0, bigBmp.Width, bigBmp.Height),
-                       System.Drawing.Imaging.ImageLockMode.ReadOnly,
-                       System.Drawing.Imaging.PixelFormat.Format24bppRgb);
-
-            int smallStride = smallData.Stride;
-            int bigStride = bigData.Stride;
-
-            int bigWidth = bigBmp.Width;
-            int bigHeight = bigBmp.Height - smallBmp.Height + 1;
-            int smallWidth = smallBmp.Width * 3;
-            int smallHeight = smallBmp.Height;
+            Rectangle rectangleResult = FindImage(smallBmp, bigBmp, tolerance);
+            if (rectangleResult != Rectangle.Empty) { return rectangleResult; }
+            else { return Rectangle.Empty; }
+        }
 
+        private static Rectangle FindImage(Bitmap needle, Bitmap haystack, double tolerance)
+        {
             Rectangle location = Rectangle.Empty;
             int margin = Convert.ToInt32(255.0 * tolerance);
 
-            unsafe
+            if (null == haystack || null == needle)
             {
-                byte* pSmall = (byte*)(void*)smallData.Scan0;
-                byte* pBig = (byte*)(void*)bigData.Scan0;
-
-                int smallOffset = smallStride - smallBmp.Width * 3;
-                int bigOffset = bigStride - bigBmp.Width * 3;
+                return location;
+            }
+            if (haystack.Width < needle.Width || haystack.Height < needle.Height)
+            {
+                return location;
+            }
 
-                bool matchFound = true;
+            var haystackArray = GetPixelArray(haystack);
+            var needleArray = GetPixelArray(needle);
+            var firstLineMatchPointsList = FindMatch(haystackArray.Take(haystack.Height - needle.Height), needleArray[0], margin);
 
-                for (int y = 0; y < bigHeight; y++)
+            var resultCollection = new ConcurrentBag<Point>();
+            ParallelLoopResult result = Parallel.ForEach<Point>(firstLineMatchPointsList, (firstLineMatchPoint) =>
```

Comment:
you are not using the result anywhere so it's no need to assigning it to a variable, also <Point> is redundant